Jenkinsfile

```diff
@@ -10,116 +14,186 @@ pipeline {
     buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
     timestamps()
   }
+  parameters {
+    string(name: 'AWSRegion', defaultValue: 'us-east-1', description: 'Enter the desired AWS region')
+    string(name: 'ECRURI', defaultValue: '054017840000.dkr.ecr.us-east-1.amazonaws.com', description: 'Enter the URI of the Container Registry')
+    string(name: 'Email', defaultValue: 'vecinomio@gmail.com', description: 'Enter the desired Email for the Job notifications')
+    booleanParam(name: 'Build', defaultValue: true, description: 'Includes Build app and Tests')
+    booleanParam(name: 'Release', defaultValue: false, description: 'Includes Tagging and Delivery')
+    booleanParam(name: 'Deployment', defaultValue: false, description: 'Deploy a new version of App')
+    booleanParam(name: 'SetNewTag', defaultValue: false, description: 'Auto-increasing version')
+    choice(name: 'AppVersion', choices: ['Minor', 'Middle', 'Major'], description: 'Pick Version Tag')
+    choice(name: 'NewVersionTrafficWeight', choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: 'Amount of traffic to the new vesion of the App')
+  }
   environment {
-    ECRURI = '054017840000.dkr.ecr.us-east-1.amazonaws.com'
+    ECRURI = "${params.ECRURI}"
+    AWSRegion = "${params.AWSRegion}"
     AppRepoName = 'snakes'
-    OPSRepoURL = 'https://github.com/IYermakov/DevOpsA3Training.git'
-    OPSRepoBranch = 'ecs-snakes'
-    Tag = ""
-    Email = 'vecinomio@gmail.com'
-    DelUnusedImage = 'docker image prune -af --filter="label=maintainer=devopsa3"'
-
+    OPSRepoURL = 'git@github.com:IYermakov/DevOpsA3Training.git'
+    OPSRepoBranch = 'weighted-tgs'
+    BuildAndTest = "${params.Build}"
+    Release = "${params.Release}"
+    Deployment = "${params.Deployment}"
+    StartVersionFrom = '1.0.0'
+    ChoiceResult = "${params.Version}"
+    CurrentVersionTrafficWeight = (10 - "${params.NewVersionTrafficWeight}".toInteger()).toString()
+    Email = "${params.Email}"
+    FailureEmailSubject = "JOB with identifier ${Tag} FAILED"
+    SuccessEmailSubject = "JOB with identifier ${Tag} SUCCESS"
   }
+
   stages {
-    stage("Build app") {
+    stage("Versioning"){
+      when { environment name: 'SetNewTag', value: 'true' }
+      steps {
+        script {
+            sh """
```

Comment:
would rewrite script to:
- update `SetNewTag` default value with `$(git describe --tags `git rev-list --tags --max-count=1` || echo ${StartVersionFrom}) `
- to not have 'Minor', 'Middle', 'Major' options in `SetNewTag`
- validate and compare current and new version, see https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash

Reply:
Done